src/fsm.jl

```diff
@@ -313,3 +313,28 @@ Return a minimal equivalent fsm.
 """
 minimize(fsm::FSM{T}) where T = (transpose ∘ determinize ∘ transpose ∘ determinize)(fsm)
 
+"""
+    union(fsm1, fsm2, ...)
+
+Merge several FSMs into a single one.
+"""
+function Base.union(fsm1::FSM{T}, fsm2::FSM{T}) where T
+    fsm = FSM{T}()
+    s = addstate!(fsm)
+    s1 = addstate!(fsm, label = "#fsm1")
+    s2 = addstate!(fsm, label = "#fsm2")
+    link!(fsm, s, s1)
+    link!(fsm, s, s2)
+    setinit!(s)
+    setfinal!(s1)
+    setfinal!(s2)
+    unionfsm = replace(fsm, Dict("#fsm1" => fsm1, "#fsm2" => fsm2))
+    foreach(states(unionfsm)) do s
+        if ! isnothing(s.label)
+            s.label = replace(s.label, "#fsm1!" => "")
+            s.label = replace(s.label, "#fsm2!" => "")
+        end
+    end
```

Comment:
I get your idea and this seems correct. However, I would prefer not to rely on the `replace` function because creating the states and links manually will be much faster and we might have to change the replace function behavior later on (especially about this special chararcter `!`).

Reply:
I see that, you already did it.